sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ReceiveLinkHandler2.java

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.implementation.handler;
+
+import com.azure.core.amqp.AmqpRetryOptions;
+import com.azure.core.amqp.implementation.AmqpMetricsProvider;
+import com.azure.core.amqp.implementation.ReactorDispatcher;
+import com.azure.core.util.logging.LoggingEventBuilder;
+import org.apache.qpid.proton.amqp.transport.DeliveryState;
+import org.apache.qpid.proton.engine.BaseHandler;
+import org.apache.qpid.proton.engine.Delivery;
+import org.apache.qpid.proton.engine.EndpointState;
+import org.apache.qpid.proton.engine.Event;
+import org.apache.qpid.proton.engine.Extendable;
+import org.apache.qpid.proton.engine.Handler;
+import org.apache.qpid.proton.engine.Link;
+import org.apache.qpid.proton.engine.Receiver;
+import org.apache.qpid.proton.message.Message;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.azure.core.amqp.implementation.ClientConstants.ENTITY_PATH_KEY;
+import static com.azure.core.amqp.implementation.ClientConstants.LINK_NAME_KEY;
+
+/**
+ * Handler that receives events from its corresponding {@link Receiver}. Handlers must be associated to a
+ * {@link Receiver} to receive its events.
+ *
+ * Note: ReceiveLinkHandler2 will become the ReceiveLinkHandler once the side by side support for v1 and v2 stack is removed.
+ *
+ * @see BaseHandler#setHandler(Extendable, Handler)
+ * @see Receiver
+ */
+public class ReceiveLinkHandler2 extends LinkHandler {
+    private final String linkName;
+
+    /**
+     * Indicates whether or not the link has ever been remotely active (ie. the service has acknowledged that we have
+     * opened a send link to the given entityPath.)
+     */
+    private final AtomicBoolean isRemoteActive = new AtomicBoolean();
+    private final AtomicBoolean isTerminated = new AtomicBoolean();
+    private final String entityPath;
+    private final ReceiverUnsettledDeliveries unsettledDeliveries;
+    private final ReceiverDeliveryHandler deliveryHandler;
+
+    /**
+     * @deprecated use {@link ReceiveLinkHandler2#ReceiveLinkHandler2(String, String, String, String, DeliverySettleMode,
+     * ReactorDispatcher, AmqpRetryOptions, boolean, AmqpMetricsProvider)} instead.
+     */
+    @Deprecated
```

Comment:
any reason to add a deprecated constructor?

Reply:
Good catch: we don't need this Ctr; let's delete it.

Reply:
This Ctr is removed.